crates/kornia-pnp/src/refine.rs

```diff
@@ -0,0 +1,333 @@
+//! Levenberg–Marquardt pose refinement for PnP solutions.
+
+use crate::pnp::PnPError;
+use glam::{Vec3, Vec3A};
+use kornia_lie::so3::SO3;
+
+// Levenberg–Marquardt defaults and numerical constants
+const DEFAULT_MAX_ITERS: usize = 20;
+const DEFAULT_EPS: f32 = 1e-6;
+const DEFAULT_LAMBDA_INIT: f32 = 1e-3;
+const DEFAULT_LAMBDA_MUL: f32 = 10.0;
+
+// Problem sizes
+const SOLVER_DIM: usize = 6; // [rx, ry, rz, tx, ty, tz]
+const JTJ_SIZE: usize = SOLVER_DIM * SOLVER_DIM; // 36
+const RESIDUAL_DIM: usize = 2; // [du, dv]
+
+// Finite-difference steps
+const FINITE_DIFF_ROT_STEP: f32 = 1e-4; // radians
+const FINITE_DIFF_TRANS_BASE_STEP: f32 = 1e-4; // world units (scaled)
+
+// Numerical thresholds
+const NUMERICAL_EPS: f32 = 1e-12; // generic tiny epsilon for stability
+const MIN_CORRESPONDENCES: usize = 3;
+const MIN_TRANSLATION_SCALE: f32 = 1.0;
+
+/// Parameters controlling the LM pose refinement.
+#[derive(Debug, Clone)]
+pub struct LMParams {
+    /// Maximum number of LM iterations.
+    pub max_iters: usize,
+    /// Convergence threshold on squared reprojection error decrease.
+    pub eps: f32,
+    /// Initial damping factor (lambda).
+    pub lambda_init: f32,
+    /// Multiplicative factor to increase/decrease lambda.
+    pub lambda_mul: f32,
+    /// Finite-difference step for rotation parameters (radians).
+    pub fd_rot_step: f32,
+    /// Base finite-difference step for translation parameters (scaled by translation magnitude).
+    pub fd_trans_base_step: f32,
+}
+
+impl Default for LMParams {
+    fn default() -> Self {
+        Self {
+            max_iters: DEFAULT_MAX_ITERS,
+            eps: DEFAULT_EPS,
+            lambda_init: DEFAULT_LAMBDA_INIT,
+            lambda_mul: DEFAULT_LAMBDA_MUL,
+            fd_rot_step: FINITE_DIFF_ROT_STEP,
+            fd_trans_base_step: FINITE_DIFF_TRANS_BASE_STEP,
+        }
+    }
+}
+
+/// Refine a pose (rvec, t) with Levenberg–Marquardt to minimize pixel reprojection error.
+///
```

Comment:
```suggestion
/// # Arguments
```
follow same convention as in other parts of the library